internal/controller/model_pod.go

```diff
@@ -71,9 +81,96 @@ func (g *PodLoadBalancerModelGenerator) findPort(subset *corev1.EndpointSubset,
 	return -1, errPortNotFoundInSubset
 }
 
+func containsPort(port int32, proto *corev1.Protocol, portList []networkingv1.NetworkPolicyPort) bool {
+	klog.Infof("Looking for port %d proto %#v", port, *proto)
+	for _, p := range portList {
+		klog.Infof("%#v: port %d proto %#v", p, p.Port.IntVal, p.Protocol)
+		if *p.Protocol == *proto &&
+			((p.EndPort == nil && p.Port.IntVal == port) ||
+				(p.EndPort != nil && p.Port.IntVal <= port && *p.EndPort >= port)) {
+			klog.Infof("Found port in %#v", p)
+			return true
+		}
+	}
+	return false
+}
+
+func buildNetworkPolicy(in *networkingv1.NetworkPolicy) model.NetworkPolicy {
+	newPolicy := model.NetworkPolicy{
+		Name:            in.Name,
+		AllowedIPBlocks: []model.AllowedIPBlock{},
+		Ports:           []model.PolicyPort{},
+	}
+	for _, ingress := range in.Spec.Ingress {
+		klog.Infof("Processing ingress %#v", ingress)
+		for _, port := range ingress.Ports {
+			newPort := model.PolicyPort{
+				Protocol: *port.Protocol,
+				EndPort:  port.EndPort,
+			}
+			if port.Port != nil {
+				newPort.Port = &port.Port.IntVal
+			}
+			klog.Infof("Adding proto %s port %d to %d",
+				newPort.Protocol, newPort.Port, newPort.EndPort)
+			newPolicy.Ports = append(newPolicy.Ports, newPort)
+		}
+		for _, from := range ingress.From {
+			if from.IPBlock == nil {
+				continue
+			}
+			newBlock := model.AllowedIPBlock{
+				Cidr: from.IPBlock.CIDR,
+			}
+			for _, except := range from.IPBlock.Except {
+				newBlock.Except = append(newBlock.Except, except)
+			}
+			klog.Infof("Adding block %s with %d excepts",
+				newBlock.Cidr, len(newBlock.Except))
+			newPolicy.AllowedIPBlocks = append(newPolicy.AllowedIPBlocks, newBlock)
+		}
```

Comment:
Could you simplify the nftables rules by detecting the case where no IP block is given (which is equivalent to allowing everything, if I'm not mistaken) and injecting 0.0.0.0/0 and 2000::/3 in such cases?

Reply:
> When a pod is isolated for ingress, the only allowed connections into the pod are those from the pod's node and those allowed by the ingress list of some NetworkPolicy that applies to the pod for ingress.

(from https://kubernetes.io/docs/concepts/services-networking/network-policies/)

My interpretation of this is: If there is an Ingress block (which there is if we arrive at line 119) but no IPBlock, then no external traffic is allowed to go to the Pod (by this policy).

Reply:
Interesting! I must say I'm really not a fan of the NetworkPolicy docs. I'll implement handling for that case.